tests/http2.rs

```diff
@@ -0,0 +1,46 @@
+mod support;
+use support::*;
+
+#[cfg(feature = "rustls-tls")]
+#[tokio::test]
+async fn http2_client_server_rustls_tls() {
+    extern crate rustls;
+
+    let server = server::http2(move |_req| async { http::Response::new("Hello".into()) });
+
+    let tls = rustls::ClientConfig::new();
```

Comment:
There is no prior knowledge configured (=at TLS level).
https://github.com/seanmonstar/reqwest/blob/9a66c5748b972ee46004963e7c365c73408140bb/src/async_impl/client.rs#L229 is missing here.